### PR TITLE
feat(components): add clear button to select combobox

### PR DESCRIPTION
**Pull request by richbachman**
## Description of the change

Adds a clear button to the select combobox. This clears the value using the `setValue` function. I think this should work with Ag-grid, but we may need to do an update to the component used there as well. 🤔 

Screen Shot 2022-05-19 at 5 01 20 PM

Asana ticket: N/A

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached.
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: packages/components/src/components/select-combobox/src/index.tsx [hunk 1]
```diff
@@ -11,50 +12,68 @@ import {
...
+              <StyledCombobox
+                state={comboboxState}
+                autoSelect
+                placeholder="Search..."
+              />
+            </InputBox>
+          </Box>
+          <SelectComboboxList state={comboboxState}>
+            {comboboxState.matches.map((itemValue, i) => (
+              <StyledComboboxItem key={itemValue + i} focusOnHover>
```
Comment (RohitRajendran) [inline packages/components/src/components/select-combobox/src/index.tsx:65]: We should remove combining the itemValue with the index for the key. It's essentially no different than just using index for the key, which isn't great.
```suggestion
              <StyledComboboxItem key={itemValue} focusOnHover>
```

**Review by RohitRajendran — COMMENTED** [review]
Unrelated to your changes but as I was testing it, I noticed that defaultValue is an optional prop for the combobox state but setting defaultValue to undefined results in the first item showing up as if "Apple" was selected. I would have expected it to be the same as if the value was "".

Could we also add some tests, por favor?

**Comment by richbachman** [comment]
> Unrelated to your changes but as I was testing it, I noticed that defaultValue is an optional prop for the combobox state but setting defaultValue to undefined results in the first item showing up as if "Apple" was selected. I would have expected it to be the same as if the value was "".
> 
> Could we also add some tests, por favor?

Tests added!

For the undefined defaultValue thing, that is by design from the primitive author. They're treating this like an html select, i.e.: 

```
<select name="pets" id="pet-select">
    <option value="dog">Dog</option>
    <option value="cat">Cat</option>
</select>
```

If no empty value option is passed it defaults tot the first option, dog.

I can see where they're coming from here, but not the biggest fan because this thing is select and combobox combined. I'm going to post a discussion thread to their repo and see what they say.

We may need to watch how we use defaultValue with the various ag-grid columns. If one ins't a string, we'll need to adjust it there.

File: packages/components/src/components/select-combobox/__tests__/select-combobox.tests.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import * as React from "react";
+import { fireEvent, render, screen } from "@testing-library/react";
+import { Default, DefaultOpen, EmptyValue } from "../stories/index.stories";
```
Comment (RohitRajendran) [inline packages/components/src/components/select-combobox/__tests__/select-combobox.tests.tsx:3]: Thanks for adding tests! Could you add one more that tests the clearing feature that you just added?

Comment (richbachman) [reply packages/components/src/components/select-combobox/__tests__/select-combobox.tests.tsx:3]: 🤦

Comment (richbachman) [reply packages/components/src/components/select-combobox/__tests__/select-combobox.tests.tsx:3]: Done!

**Review by RohitRajendran — APPROVED** [review]
:shipit:

**Pull request merged**